pkg/controller/service_control.go

```diff
@@ -130,6 +133,53 @@ func (c *realServiceControl) recordServiceEvent(verb, name, kind string, object
 	}
 }
 
+func (c *realServiceControl) SyncComponentService(tc runtime.Object, newSvc, oldSvc *corev1.Service, syncClusterIPFn func(newSvc *corev1.Service)) (*corev1.Service, error) {
```

Comment:
It seems that, the function parameter `syncClusterIPFn` is only copies `.spec.clusterIP` from old svc object to the new one, it seems a function modifier would be too generic for that purpose. Maybe we could use a boolean to differentiate that? And we wouldn't need to pass a function every time invoking this method, what't do you think?

Reply:
@hanlins yes, I think that make sense, will make the change